src/Aspire.Hosting/Dcp/ApplicationExecutor.cs

```diff
@@ -1878,4 +1881,60 @@ private string ReplaceLocalhostWithContainerHost(string value)
                     .Replace("127.0.0.1", hostName)
                     .Replace("[::1]", hostName);
     }
+
+    /// <summary>
+    /// Create a patch update using the specified resource.
+    /// A copy is taken of the resource to avoid permanently changing it.
+    /// </summary>
+    internal static V1Patch CreatePatch<T>(T obj, Action<T> change) where T : CustomResource
+    {
+        // This method isn't very efficient.
+        // If mass or frequent patches are required then we may want to create patches manually.
+        var current = JsonSerializer.SerializeToNode(obj);
+
+        var copy = JsonSerializer.Deserialize<T>(current)!;
+        change(copy);
+
+        var changed = JsonSerializer.SerializeToNode(copy);
+
+        var jsonPatch = current.CreatePatch(changed);
+        return new V1Patch(jsonPatch, V1Patch.PatchType.MergePatch);
+    }
+
+    internal async Task StopResourceAsync(string resourceName, CancellationToken cancellationToken)
+    {
+        var matchingResource = _appResources.SingleOrDefault(r => string.Equals(r.DcpResource.Metadata.Name, resourceName, StringComparisons.ResourceName));
+        if (matchingResource == null)
+        {
+            throw new InvalidOperationException($"Resource '{resourceName}' not found.");
+        }
+
+        if (matchingResource.DcpResource is Container c)
+        {
+            var patch = CreatePatch(c, obj => obj.Spec.Stop = true);
+            await kubernetesService.PatchAsync(c, patch, cancellationToken).ConfigureAwait(false);
+        }
+        else if (matchingResource.DcpResource is Executable e)
+        {
+            var patch = CreatePatch(e, obj => obj.Spec.Stop = true);
+            await kubernetesService.PatchAsync(e, patch, cancellationToken).ConfigureAwait(false);
+        }
+        else if (matchingResource.DcpResource is ExecutableReplicaSet rs)
+        {
+            var patch = CreatePatch(rs, obj => obj.Spec.Stop = true);
+            await kubernetesService.PatchAsync(rs, patch, cancellationToken).ConfigureAwait(false);
+        }
+        else
+        {
+            throw new InvalidOperationException($"Unexpected resource type: {matchingResource.DcpResource.GetType().FullName}");
```

Comment:
What is the experience when this throws?

Reply:
I don't think there is any. Just being defensive.